spec/inverted-index-test.js

```diff
@@ -0,0 +1,200 @@
+        expect(theIndex.getIndex(docs)).not.toBe('JSON file is empty');
+      });
+
+      it('Should have the getIndex method defined', () => {
+        expect(theIndex.getIndex).toBeDefined();
+      });
+    });
+
+    describe('Populate Index', () => {
+      it('Should verify Index is created', () => {
+        //this verifies that an entry for a word is created
+        expect(allIndexes['books.json'].alice).toBeDefined();
+      });
+
+      it('Should map words to correct docs', () => {
+        expect(allIndexes['books.json'].alice).toEqual([1]);
+        expect(allIndexes['books.json'].a).toEqual([1, 2, 3]);
+      });
+
+      //Should ensure a new index is not overwritten
+      //This will be tested in two ways
+      it('Should ensure a new index does not overwrite other indexes', () => {
+        //check that allIndexes Object has number of keys greater than 1
+        expect(Object.keys(allIndexes).length).toBeGreaterThan(1);
+      });
+    });
+
+    describe('Search index', () => {
+      it('Should have searchIndex method defined', () => {
+        expect(theIndex.searchIndex).toBeDefined();
+      });
+
+      it('Should return the correct results', () => {
+        expect(theIndex.searchIndex('books.json','ring')).toEqual({'ring':[2]});
+      });
+
+      it('Should return the correct results', () => {
+        expect(theIndex.searchIndex('books.json', 'an')).toEqual({'an':[2, 4]});
+      });
+
+      it('Should return correct results for searching all documents', () => {
+        expect(theIndex.searchIndex('books.json', multSearchTerms)).toEqual({
+          'an': [2, 4],
+          'alice': [1],
+          'man': [2],
+          'tormented': [3],
+          'market': 'Word not found!'
+        });
+      });
+
+      it('Should return correct results when filename is not specified', () => {
+        expect(theIndex.searchIndex(allSearchTerms)).toEqual(
+          { 'books.json':
+              { powerful: [ 2 ],
+                dangerous: 'Word not found!',
+                young: [ 3 ],
+                spy: 'Word not found!',
+                wizard: [ 2 ],
+                elf: [ 2 ] },
+              'books_2.json':
+              { powerful: [ 1 ],
+                dangerous: [ 2 ],
+                young: [ 1 ],
+                spy: [ 2 ],
+                wizard: [ 1 ],
+                elf: 'Word not found!' },
+              'empty_doc.json':
+              { powerful: 'Word not found!',
+                dangerous: 'Word not found!',
+                young: 'Word not found!',
+                spy: 'Word not found!',
+                wizard: 'Word not found!',
+                elf: 'Word not found!' },
+              'invalid_doc.json':
+              { powerful: 'Word not found!',
+                dangerous: 'Word not found!',
+                young: 'Word not found!',
+                spy: 'Word not found!',
+                wizard: 'Word not found!',
+                elf: 'Word not found!' }}
```

Comment:
It is best to have this object defined in a variable outside your it block, probably at the top of your code.